lib/settings.rb

```diff
@@ -5,7 +5,13 @@ def self.all
       if ENV['ENV'] == 'test'
         YAML.load_file(File.join('.', 'spec/config.test.yml'))
       else
-        YAML.load_file(File.join('.', 'config.yml'))
+        # check if the first command line parameter contains a value and use that for the config file
```

Comment:
I was thinking about the code a bit and what you could do is to extract all the config-file related logic to a method and then call it. Could be something like this:

```ruby
# Reads the settings from the yaml file
class Settings
  def self.all
    @settings ||= begin
      YAML.load_file(File.join('.', config_path))
    end
  end

  def self.config_path
    return "spec/config.test.yml" if ENV['ENV'] == 'test'
    return "config/#{ARGV[0]}.yml" if ARGV[0]
    "config.yml"
  end
end
```

The `YAML.load_file` should already throw an understandable error, so I don't think we would need to handle it. Let me know what you think

Reply:
Same as above - great idea but no idea how to do this in Ruby. Will have to look into it.